src/Toolkit/Str.php

```diff
@@ -481,13 +481,21 @@ public static function float(
 		// make sure $value is not null
 		$value ??= '';
 
+		// turn the value into a string
+		$value = (string)$value;
+
 		// Convert exponential to decimal, 1e-8 as 0.00000001
 		if (strpos(strtolower($value), 'e') !== false) {
 			$value = rtrim(sprintf('%.16f', (float)$value), '0');
 		}
 
 		$value   = str_replace(',', '.', $value);
-		$decimal = strlen(substr(strrchr($value, '.'), 1));
+		$decimal = strrchr($value, '.');
+		$decimal = match ($decimal) {
+			false   => 0,
+			default => strlen(substr($decimal, 1))
```

Comment:
This can be shortened to:

```suggestion
			default => strlen($decimal) - 1
```